### ISSUE TITLE
"Segmentation fault (core dumped)" when using string timestamp for x-axis

### ISSUE THREAD

**Issue by shanemmattner** [predict-idlab/plotly-resampler#153]
I'm trying to use `plotly-resampler` with time-stamp data.  This data works fine in a normal plotly figure:

```
SN,TS,SENSOR,READING
0004,2022-12-19T16:08:08Z,sensor1,1.992
0004,2022-12-19T16:08:09Z,sensor1,1.992
0004,2022-12-19T16:08:09Z,sensor1,1.992
0004,2022-12-19T16:08:10Z,sensor1,1.992
0004,2022-12-19T16:08:11Z,sensor1,1.992
0004,2022-12-19T16:08:12Z,sensor1,1.992
0004,2022-12-19T16:08:13Z,sensor1,1.992
0004,2022-12-19T16:08:14Z,sensor1,1.993
...
```

But when I try the same figure with plotly-resampler I get `Segmentation fault (core dumped)`.  If I replace the `x-axis` with integers the resampler works fine:

```python
x = np.arange(len(df['READING']))
f.add_trace(go.Scattergl(x=x,y=df['READING'], mode='lines+markers', yaxis="y1", name='reading'))
# Seg fault code below
# f.add_trace(go.Scattergl(x=df['TS'],y=df['READING'], mode='lines+markers', yaxis="y1", name='reading'))


```

**Comment by shanemmattner** [comment]
When I convert this string to `datetime` the issue is resolved.  Is this an issue to fix on `plotly-resampler` side?

```python
df['TS'] = df['TS'].apply(lambda x : datetime.strptime(x, '%Y-%m-%dT%H:%M:%S%z'))
```

**Comment by jonasvdd** [comment]
Hi @shanemmattner!

Thank you for pointing this issue out! 
I think your issue is highly related to #116 &  #115, which is integrated in `plotly-resampler>=0.8.3`. 
Could you look which plotly-resampler version you have installed (and retest with the last version of plotly-resampler)?

This code snippet (which is introduced in `plotly-resampler>0.8.3`) tries to convert the `str` dtype to pd.datetime.
https://github.com/predict-idlab/plotly-resampler/blob/a01f0bcfe6763aea823dbac65b2de7b78775ac8b/plotly_resampler/figure_resampler/figure_resampler_interface.py#L698-L716

Happy holidays 🎅🏼 🎉

**Comment by shanemmattner** [comment]
Problematic version: `Version: 0.8.1`

I confirmed that the issue is fixed by upgrading to `0.8.3`.  Thank you @jonasvdd!
